flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/api/SavepointWindowReaderITCase.java

```diff
@@ -348,24 +359,30 @@ public void testApplyWindowStateReader() throws Exception {
     public void testApplyEvictorWindowStateReader() throws Exception {
         String savepointPath =
                 takeSavepoint(
-                        numbers,
-                        source -> {
+                        WaitingWindowAssigner.wrap(
+                                TumblingEventTimeWindows.of(Time.milliseconds(10))),
+                        (windowAssigner) -> {
                             StreamExecutionEnvironment env =
                                     StreamExecutionEnvironment.getExecutionEnvironment();
                             env.setStateBackend(getStateBackend());
                             env.setParallelism(4);
 
-                            env.addSource(source)
-                                    .rebalance()
-                                    .assignTimestampsAndWatermarks(
-                                            WatermarkStrategy.<Integer>noWatermarks()
-                                                    .withTimestampAssigner((event, timestamp) -> 0))
-                                    .keyBy(id -> id)
-                                    .window(TumblingEventTimeWindows.of(Time.milliseconds(10)))
-                                    .evictor(new NoOpEvictor<>())
-                                    .apply(new NoOpWindowFunction())
-                                    .uid(uid)
-                                    .addSink(new DiscardingSink<>());
+                            try {
+                                env.addSource(createSource(numbers))
+                                        .rebalance()
+                                        .assignTimestampsAndWatermarks(
+                                                WatermarkStrategy.<Integer>noWatermarks()
+                                                        .withTimestampAssigner(
+                                                                (event, timestamp) -> 0))
+                                        .keyBy(id -> id)
+                                        .window(windowAssigner)
+                                        .evictor(new NoOpEvictor<>())
+                                        .apply(new NoOpWindowFunction())
+                                        .uid(uid)
+                                        .addSink(new DiscardingSink<>());
+                            } catch (Exception e) {
+                                e.printStackTrace();
+                            }
```

Comment:
Why are exceptions not logged via the logging framework?

Reply:
I also wonder if we shouldn't rather fail the test if an exception is thrown while creating the job?

Reply:
Make it works to change the signature of takeSavepoint() to use `FunctionWithException` instead of `Function`?

Reply:
Thanks a lot!